source/grid/tria_accessor.cc

```diff
@@ -1307,7 +1307,7 @@ namespace
     // volume relative to |v01|*|v02|*|v03|. the test checks the
     // squares of these to avoid taking norms/square roots:
     if (std::abs((v03 * normal) * (v03 * normal) /
-                 ((v03 * v03) * (v01 * v01) * (v02 * v02))) >= 1e-24)
+                 ((v03 * v03) * (v01 * v01) * (v02 * v02))) >= 1e-10)
```

Comment:
I thought about this yesterday. The `vij` are all differences in vertex coordinates. So they are of size `h`. The `normal` is a cross product, so it's of size `h^2`. The expression in question here is therefore of size `O(h^6)/O(h^6)=O(1)`. I think it doesn't make sense to compare such a quantity to `1e-24`: Even just round-off of exact quantities would satisfy the inequality.

I have not tried to follow what size the error between a planar approximation of a face (the `else` branch) and the exact formula here (the `if` branch) has, but I think the `1e-24` is unreasonable :-)

Reply:
Still, I tend to disagree - we expect the number `v03 * normal` to be small compared to `h^3`, say `epsilon * h^3`. Thus, the expression in question is `epsilon^2 * h^6 / h^6 = epsilon^2`. The fact that you scale by `h` should not matter here.

That aside, another thing I do not understand is why the comment is afraid of square roots but then uses an unnecessary division - the denominator could easily be moved to the right hand side. (Well, it's probably because of the three square roots.)

Reply:
In order to make the proposal for 1e-10, I made a more simplistic-practical test. I wrote the triangulation to an msh file by using gridout, for which I was sure that all faces are perfectly planar, and then read it back in. Then I printed this number for most of the faces of the read-in triangulation. It was somewhere between 1.e-13 and 1.e-12, which obviously comes from the roundoff error of writing the nodes on an ascii file in the standard format used by deal.II with the default options. So Iguess making the tolerance 1.e-10 gives a safe margin to not get into these errros.

Reply:
@kronbichler We should come to a conclusion about this issue here before the release. I tend to agree that we can't promise more accuracy than the accuracy with which we generally read data from input files. I think it seems reasonable to say that if, to 6 digits of accuracy, four vertices are in a plane, then we call them "in a plane". At least one should assume that in that case, the *intent* was to have things in a plane.

So I think I'm ok with the patch as is. @kronbichler ?